staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go

```diff
@@ -94,63 +166,71 @@ func GRPCVerbosity(level int) Option {
 	}
 }
 
-// RegistrarSocketPath sets the file path for a Unix domain socket.
-// If RegistrarListener is not used, then Start will remove
-// a file at that path, should one exist, and creates a socket
-// itself. Otherwise it uses the provided listener and only
-// removes the socket at the specified path during shutdown.
+// RegistrarDirectoryPath sets the path to the directory where the kubelet
+// expects to find registration sockets of plugins. Typically this is
+// /var/lib/kubelet/plugins_registry with /var/lib/kubelet being the kubelet's
+// data directory.
 //
-// At least one of these two options is required.
-func RegistrarSocketPath(path string) Option {
+// This is also the default. Some Kubernetes clusters may use a different data directory.
+// This path must be the same inside and outside of the driver's container.
+// The directory must exist.
+func RegistrarDirectoryPath(path string) Option {
 	return func(o *options) error {
-		o.pluginRegistrationEndpoint.path = path
+		o.pluginRegistrationEndpoint.dir = path
 		return nil
 	}
 }
 
-// RegistrarListener sets an already created listener for the plugin
-// registration API. Can be combined with RegistrarSocketPath.
+// RegistrarSocketName sets the name of the socket inside the directory where
+// the kubelet watches for registration sockets (see RegistrarDirectoryPath).
 //
-// At least one of these two options is required.
-func RegistrarListener(listener net.Listener) Option {
+// Usually DRA drivers should not need this option. It is provided to
+// support updates from an installation which used an older release of
+// of the helper code.
+//
+// The default is <driver name>-reg.sock. When rolling updates are enabled (not supported yet),
+// it is <driver name>-<uid>-reg.sock.
+func RegistrarSocketName(name string) Option {
```

Comment:
Is `RegistrarSocketFilename` more accurate?

Also, who is responsible for determining the `uuid` mentioned in the docs?

Reply:
OK. Thanks. That's clearer. Note that the docstring refers to `RegistrarSocketName` which we have renamed, correct?

Reply:
I also renamed it in https://github.com/kubernetes/kubernetes/pull/129832.